test/TestProjects/RequestOptionsAllOptional-LowLevel/Generated/RequestOptionsAllOptionalClient.cs

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace RequestOptionsAllOptional_LowLevel
+{
+    /// <summary> The RequestOptionsAllOptional service client. </summary>
+    public partial class RequestOptionsAllOptionalClient
+    {
+        private const string AuthorizationHeader = "Fake-Subscription-Key";
+        private readonly AzureKeyCredential _keyCredential;
+
+        private readonly HttpPipeline _pipeline;
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly Uri _endpoint;
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual HttpPipeline Pipeline { get => _pipeline; }
+
+        /// <summary> Initializes a new instance of RequestOptionsAllOptionalClient for mocking. </summary>
+        protected RequestOptionsAllOptionalClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of RequestOptionsAllOptionalClient. </summary>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> is null. </exception>
+        public RequestOptionsAllOptionalClient(AzureKeyCredential credential, Uri endpoint = null, RequestOptionsAllOptionalClientOptions options = null)
+        {
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+            endpoint ??= new Uri("http://localhost:3000");
+
+            options ??= new RequestOptionsAllOptionalClientOptions();
+
+            _clientDiagnostics = new ClientDiagnostics(options);
+            _keyCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() }, new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
+            _endpoint = endpoint;
+        }
+
+        /// <summary> No RequestBody and ResponseBody. </summary>
+        /// <param name="id"> Query parameter Id. </param>
+        /// <param name="top"> Query parameter top. </param>
+        /// <param name="skip"> Query parameter skip. </param>
+        /// <param name="status"> Query parameter status. </param>
+        /// <param name="options"> The request options. </param>
+#pragma warning disable AZC0002
+        public virtual async Task<Response> NoRequestBodyResponseBodyAsync(int id, int? top = null, int skip = 12, string status = "start", RequestOptions options = null)
```

Comment:
@ShivangiReja, help me understand here ... in this case RequestOptions is optional because the "make all RequestOptions optional" flag has been set to true?  And that happens in [Configuration.json](https://github.com/Azure/autorest.csharp/pull/1555/files/081a4f571ec278984caaf42a9e0d724a544fc98f#diff-fe472060d1fb957ceb6443d34446fe9ffec8992d1e8d6453ba703f84f67d4fb8R12)?